libsql/internal/hrana/batch_result.go

```diff
@@ -1,6 +1,51 @@
 package hrana
 
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
 type BatchResult struct {
-	StepResults []*StmtResult `json:"step_results"`
-	StepErrors  []*Error      `json:"step_errors"`
+	StepResults      []*StmtResult `json:"step_results"`
+	StepErrors       []*Error      `json:"step_errors"`
+	ReplicationIndex *uint64       `json:"replication_index"`
+}
+
+func (b *BatchResult) UnmarshalJSON(data []byte) error {
+	type Alias BatchResult
+	aux := &struct {
+		ReplicationIndex interface{} `json:"replication_index,omitempty"`
+		*Alias
+	}{
+		Alias: (*Alias)(b),
+	}
+
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	if aux.ReplicationIndex == nil {
+		b.ReplicationIndex = nil
```

Comment:
Isn't the value equal to nil by default?

Reply:
here and on rest of the similar comments: yes the value would be nil by default and does not require assigning. I will update the PR